### PR TITLE
Errors when indexing the testnet

### LINKED ISSUE #1937: Errors when indexing the testnet
### Actual behavior

```
2023-11-23T00:42:11.371 ERROR [blocks] Error occurred when normalizing block at height undefined, id: undefined:
Error: Undefined binding(s) detected when compiling SELECT. Undefined column(s): [height] query: select `numberOfEvents`, `reward` from `blocks` where `height` = ? limit ?
    at QueryCompiler_MySQL.toSQL (/home/lisk/lisk-service/blockchain-indexer/node_modules/knex/lib/query/querycompiler.js:112:13)
    at Builder.toSQL (/home/lisk/lisk-service/blockchain-indexer/node_modules/knex/lib/query/querybuilder.js:84:44)
    at Object.find (/home/lisk/lisk-service/blockchain-indexer/node_modules/lisk-service-framework/src/database/util.js:491:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /home/lisk/lisk-service/blockchain-indexer/shared/dataService/business/blocks.js:87:25
    at async normalizeBlock (/home/lisk/lisk-service/blockchain-indexer/shared/dataService/business/blocks.js:86:38)
    at async formatBlock (/home/lisk/lisk-service/blockchain-indexer/shared/dataService/blocks.js:88:19)
    at async newBlockProcessor (/home/lisk/lisk-service/blockchain-indexer/shared/messageProcessor.js:80:19)
    at async Queue.<anonymous> (/home/lisk/lisk-service/blockchain-indexer/shared/messageProcessor.js:146:4)
    at async /home/lisk/lisk-service/blockchain-indexer/node_modules/bull/lib/p-timeout.js:69:17
...
...
23-11-23T06:27:38.773 INFO [indexStatus] currentChainHeight: undefined, lastIndexedBlockHeight: undefined
2023-11-23T06:27:38.773 INFO [indexStatus] Block index status: undefined/undefined blocks indexed (undefined%).
2023-11-23T06:27:38.958 INFO [updateAccounts] Triggered account balance updates successfully.
2023-11-23T06:27:41.384 WARN [indexStatus] Error while checking index readiness: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2023-11-23T06:27:48.958 INFO [updateAccounts] Triggered account balance updates successfully.
2023-11-23T06:27:51.633 WARN [indexStatus] Error while checking index readiness: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2023-11-23T06:27:53.773 WARN [indexStatus] Error while checking index readiness: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2023-11-23T06:27:53.773 INFO [indexStatus] currentChainHeight: undefined, lastIndexedBlockHeight: undefined
2023-11-23T06:27:53.773 INFO [indexStatus] Block index status: undefined/undefined blocks indexed (undefined%).
2023-11-23T06:27:58.950 INFO [updateAccounts] Triggered account balance updates successfully.
2023-11-23T06:28:00.467 ERROR [blocks] Error occurred when normalizing block at height undefined, id: undefined:
KnexTimeoutError: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
    at Client_MySQL2.acquireConnection (/home/lisk/lisk-service/blockchain-indexer/node_modules/knex/lib/client.js:332:26)
    at runNextTicks (node:internal/process/task_queues:60:5)
    at process.processTimers (node:internal/timers:509:9)
    at async Transaction_MySQL2.acquireConnection (/home/lisk/lisk-service/blockchain-indexer/node_modules/knex/lib/execution/transaction.js:258:28)
2023-11-23T06:28:01.081 WARN [indexStatus] Error while checking index readiness: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2023-11-23T06:28:08.766 WARN [indexStatus] Error while checking index readiness: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2023-11-23T06:28:08.766 INFO [indexStatus] currentChainHeight: undefined, lastIndexedBlockHeight: undefined
2023-11-23T06:28:08.766 INFO [indexStatus] Block index status: undefined/undefined blocks indexed (undefined%).
2023-11-23T06:28:08.952 INFO [updateAccounts] Triggered account balance updates successfully.
2023-11-23T06:28:08.953 WARN [BROKER] Request 'indexer.blocks' is timed out.,[object Object]
2023-11-23T06:28:10.001 WARN [BROKER] Request 'indexer.status' is timed out.,[object Object]
2023-11-23T06:28:11.659 WARN [indexStatus] Error while checking index readiness: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
```

### Expected behavior

Lisk Service should be able to index without any errors.

### Steps to reproduce

- Run Lisk Service against testnet 

### Which version(s) does this affect? (Environment, OS, etc...)

release/0.7.0 (commit: 4735a6dbf1f8bc1bcbe9e04c19b8975bc9b1897e)

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?

This PR resolves #1937 #1943

### How was it solved?
- [x] Start block processing queues after indexer start to avoid logs
- [x] Update missing blocks job
    - [x] Update frequency
    - [x] Remove scheduling from indexer based on height gap findings
    - [x] Fetch index verified height from blocks table and drop key value table key.
- [x] Fix blockchain connector crash
    - [x] on new block event
    - [x] on sqlite cache read failure
- [x] Fix error log in indexer from indexing queues due to app not instantiated

### How was it tested?
- [x] Local

### CODE REVIEW

File: services/blockchain-indexer/shared/indexer/blockchainIndex.js [hunk 1]
```diff
@@ -850,9 +849,19 @@ const addHeightToIndexBlocksQueue = async (height, priority) => {
...
+		{
+			isFinal: true,
+			sort: 'height:desc',
+			limit: 1,
+		},
+		['height'],
+	);
 
-const getIndexVerifiedHeight = () => keyValueTable.get(INDEX_VERIFIED_HEIGHT);
+	 return lastIndexedFinalBlock.height || getGenesisHeight();
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/indexer/blockchainIndex.js:863]: Should return null if nothing is indexed.

```suggestion
	 return lastIndexedFinalBlock.height || null;
```

File: services/blockchain-coordinator/shared/scheduler.js [hunk 2]
```diff
@@ -94,7 +93,7 @@ const waitForGenesisBlockIndexing = resolve =>
 				return resolve(true);
 			}
 
-			if (jobCount <= 1) {
+			if (jobCount >= 1) {
```
Comment (sameersubudhi) [inline services/blockchain-coordinator/shared/scheduler.js:96]:
```suggestion
			if (jobCount <= 1) {
```

File: services/blockchain-coordinator/shared/scheduler.js [hunk 3]
```diff
@@ -121,6 +120,8 @@ const scheduleBlocksIndexing = async heights => {
 
 	const isMultiBatch = numBatches > 1;
 	for (let i = 0; i < numBatches; i++) {
+		await waitForJobCountToFallBelowThreshold();
```
Comment (sameersubudhi) [inline services/blockchain-coordinator/shared/scheduler.js:123]:
```suggestion
```

File: services/blockchain-coordinator/shared/scheduler.js [hunk 4]
```diff
@@ -137,7 +138,6 @@ const scheduleBlocksIndexing = async heights => {
 					0,
 				)} - ${blockHeightsBatch.at(-1)}, ${blockHeightsBatch.length} blocks).`,
 			);
```
Comment (sameersubudhi) [inline services/blockchain-coordinator/shared/scheduler.js:140]:
```suggestion
			);
		await waitForJobCountToFallBelowThreshold();
```

File: services/blockchain-coordinator/shared/scheduler.js [hunk 5]
```diff
@@ -121,6 +120,8 @@ const scheduleBlocksIndexing = async heights => {
 
 	const isMultiBatch = numBatches > 1;
 	for (let i = 0; i < numBatches; i++) {
+		await waitForJobCountToFallBelowThreshold();
+
```
Comment (sameersubudhi) [inline services/blockchain-coordinator/shared/scheduler.js:124]:
```suggestion
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/blockchain-coordinator/shared/scheduler.js (hunk 3 above)
Comment (priojeetpriyom) [reply services/blockchain-coordinator/shared/scheduler.js:123]: Discussed offline.

File: services/blockchain-coordinator/shared/scheduler.js (hunk 4 above)
Comment (priojeetpriyom) [reply services/blockchain-coordinator/shared/scheduler.js:140]: Discussed offline.

File: services/blockchain-coordinator/shared/scheduler.js (hunk 2 above)
Comment (priojeetpriyom) [reply services/blockchain-coordinator/shared/scheduler.js]: Applied.

File: services/blockchain-indexer/shared/indexer/blockchainIndex.js (hunk 1 above)
Comment (priojeetpriyom) [reply services/blockchain-indexer/shared/indexer/blockchainIndex.js]: Applied.

**Review by sameersubudhi — APPROVED** [review]

**Pull request merged**